src/history/HistoryManagerImpl.cpp

```diff
@@ -251,7 +242,18 @@ HistoryManagerImpl::maybeQueueHistoryCheckpoint()
 void
 HistoryManagerImpl::queueCurrentHistory()
 {
-    auto has = getLastClosedHistoryArchiveState();
+    // Subtle: depending on the protocol version,
```

Comment:
related to my previous comment: I don't see why we need this here.
Basically: to queue things up we should never have to wait.

Later (actual publish), we can tell if we need to wait in `ResolveSnapshotWork` directly by for example seeing that we have shadows or not.